bin/faucet/src/client.rs

```diff
@@ -200,7 +200,7 @@ impl FaucetClient {
 pub async fn initialize_faucet_client(
     config: &FaucetConfig,
 ) -> Result<(ApiClient<Channel>, BlockHeader, ChainMmr), ClientError> {
-    let endpoint = tonic::transport::Endpoint::try_from(config.node_url.clone())
+    let endpoint = tonic::transport::Endpoint::try_from(config.node_url.to_string())
```

Comment:
If we use `http::uri::Uri` instead of `url::Url`, we could rely on `From<Uri> for Endpoint` here

https://docs.rs/http/latest/http/uri/index.html

Reply:
Ah. And `tonic` actually [re-exports](https://docs.rs/tonic/latest/tonic/transport/struct.Uri.html) this I think?

Okay so probably we should go with that. Feels a bit weird because URIs allow for more. But I guess if the user wants to type in wrong things that's on him :)

Reply:
Would we use `Uri` for all endpoints then? This is only really useful for the client side of things though. The server side still wants to `TcpListener::bind` which for `Uri` implies casting it back to `str` afaiu 🤔